test/unittests/commands/ingest_xmlupload/test_user_information.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
+
+from pathlib import Path
+
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+from dsp_tools.commands.ingest_xmlupload import user_information
+from dsp_tools.commands.ingest_xmlupload.user_information import IngestInformation
+
+user_information.csv_filepath = Path()
+user_information.maximum_prints = 1
+
+
+class TestIngestInformation:
+    def test_no_problems(self) -> None:
+        expected = (
+            "All media referenced in the XML file were uploaded to sipi.\n"
+            "No media was uploaded to sipi that was not referenced in the XML file."
+        )
+        assert IngestInformation([], []).all_good_msg() == expected
+
+    def test_not_all_good(self) -> None:
+        assert not IngestInformation(["unused_media"], []).all_good_msg()
+
+    def test_unused_media(self) -> None:
+        expected = (
+            "The upload cannot continue as there are problems with the media referenced in the XML.\n"
+            "    The following media were uploaded to sipi but not referenced in the data XML file:\n"
+            "    - unused_media"
+        )
+        res_msg = IngestInformation(["unused_media"], [])._get_error_msg()
+        assert res_msg == expected
+
+    def test_not_uploaded(self) -> None:
+        expected = (
+            "The upload cannot continue as there are problems with the media referenced in the XML.\n"
+            "    The following media were not uploaded to sipi but referenced in the data XML file:\n"
+            "    - Resource ID: 'no upload id' | Filepath: 'media path'"
+        )
+        res_msg = IngestInformation([], [("no upload id", "media path")])._get_error_msg()
+        assert res_msg == expected
+
+    def test_all_problem_with_df_msg(self) -> None:
+        expected = (
+            "The upload cannot continue as there are problems with the media referenced in the XML.\n"
+            "    Media was uploaded to Sipi which was not referenced in the XML file.\n"
+            "    The file 'UnusedMediaUploadedInSipi.csv' was saved in '.' with the filenames.\n\n"
+            "    Media was referenced in the XML file but not previously uploaded to sipi:\n"
+            "    The file 'NotUploadedFilesToSipi.csv' was saved in '.' with the resource IDs and filenames."
```

Comment:
For me, this message is not clear. Perhaps we discuss it together in person how to reformulate?